### PR TITLE
Added convenience methods for MorphismBetweenDirectSums and DirectSum(Op)

### PR DESCRIPTION
**Pull request by mohamed-barakat**

### CODE REVIEW

File: CAP/gap/UniversalObjects.gd [hunk 1]
```diff
@@ -1098,6 +1098,9 @@ DeclareOperation( "AddInitialObjectFunctorial",
 DeclareOperationWithCache( "DirectSumOp",
                            [ IsList, IsCapCategoryObject ] );
 
+DeclareOperation( "DirectSumOp",
```
Comment (sebasguts) [inline CAP/gap/UniversalObjects.gd]: How is this function supposed to be called? Do you call it explicitly with a category? Then one might want to name it DirectSum.

Comment (sebastianpos) [inline CAP/gap/UniversalObjects.gd]: DirectSum is a global function in Gap that automatically redirects to DirectSumOp.

Comment (sebasguts) [inline CAP/gap/UniversalObjects.gd]: Okay. So one has to call DirectSumOp on empty lists, giving the category.

Comment (sebastianpos) [inline CAP/gap/UniversalObjects.gd]: Actually you can call `DirectSum( category, list_of_objects )`,
which redirects to `DirectSumOp( list_of_objects, category )`.

Comment (sebasguts) [inline CAP/gap/UniversalObjects.gd]: I see :)

File: CAP/gap/UniversalObjects.gi [hunk 2]
```diff
@@ -421,6 +438,20 @@ InstallMethod( MorphismBetweenDirectSums,
 end );
 
 ##
+InstallMethod( MorphismBetweenDirectSums,
+               [ IsCapCategoryObject, IsList, IsCapCategoryObject ],
+               
+  function( S, morphism_matrix, T )
+    
+    if IsZero( S ) or IsZero( T ) then
```
Comment (sebastianpos) [inline CAP/gap/UniversalObjects.gi]: Depending on the category this test might be way too expensive
(e.g. in the case of sheaves on toric varieties).
Doesn't it suffice to check if the given list is either empty
or a list containing only empty lists?

Comment (sebasguts) [inline CAP/gap/UniversalObjects.gi]: The added implementation of `DirectSumOp` suggests that this is the use-case, and I also think this should be the way to do it. E.g., for modules, @sebastianpos way would produce the morphism one might have wanted to produce, whereas this implementation would construct a matrix made of `0`s.

Comment (mohamed-barakat) [inline CAP/gap/UniversalObjects.gi]: Thanks, addressed in a650b47.

**Pull request merged**